KPV-3339_Allow_register_with_2_letters_cifs

PedroR-dev opened the pull request:
Updated regex to met task requierements

PedroR-dev committed:
Update nid regex in association register flow to met task requirements

iduetxe commented on src/groovy/kuorum/web/commands/profile/funnel/FunnelFillBasicDataCommand.groovy:
```diff
@@ -41,7 +41,7 @@ class FunnelFillBasicDataCommand {
         phonePrefix nullable: false
         phone nullable: false, matches: "^[0-9]{9}\$"
-        nid nullable: false, matches: "^[A-Z][0-9]{8}", validator: { val, obj ->
+        nid nullable: false, matches: "^(?:[A-Z]\\d{7}[A-Z]|[A-Z]\\d{8}|\\d{8}[A-Z])\$", validator: { val, obj ->
```
Not alowed CIF with only a letter at the end of the number

PedroR-dev committed:
Changed regex to uncover non-desirable cifs.
Requirements:
Allowed letters: CDEFGHJKLMNPQRSUVW
Allowed patterns:
Letter + 7 digits + Letter
 OR
Letter + 8 digits
